config/connections.go

```diff
@@ -379,11 +379,18 @@ func LocalConnect(ctx context.Context,
 }
 
 // HonoConnect connects to the Hub.
-func HonoConnect(sigs <-chan os.Signal,
+func HonoConnect(sigs chan os.Signal,
 	statusPub message.Publisher,
 	honoClient *conn.MQTTConnection,
 	logger logger.Logger,
 ) error {
+
+	if sigs == nil {
```

Comment:
It seems that this is not a part of: Paho credentials provider. Could you please move it to a new bug?